domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/PerformanceMetricsLogger.kt

```diff
@@ -0,0 +1,260 @@
+    firstTimestamp = oppiaClock.getCurrentTimeMs()
+  }
+
+  /** Logs the apk size of the application as a performance metric. */
+  fun logApkSize() {
+    performanceMetricsController.logLowPriorityMetricEvent(
+      oppiaClock.getCurrentTimeMs(),
+      OppiaMetricLog.CurrentScreen.HOME_SCREEN,
+      createApkSizeLoggableMetric(performanceMetricsUtils.getApkSize())
+    )
+  }
+
+  /** Logs the storage usage of the application as a performance metric. */
+  fun logStorageUsage() {
+    performanceMetricsController.logLowPriorityMetricEvent(
+      oppiaClock.getCurrentTimeMs(),
+      OppiaMetricLog.CurrentScreen.HOME_SCREEN,
+      createStorageUsageLoggableMetric(performanceMetricsUtils.getUsedStorage())
+    )
+  }
+
+  /** Logs the startup latency of the application as a performance metric. */
+  fun logStartupLatency() {
+    val startupLatency = oppiaClock.getCurrentTimeMs() - firstTimestamp
+    if (startupLatency >= 0) {
+      performanceMetricsController.logLowPriorityMetricEvent(
+        oppiaClock.getCurrentTimeMs(),
+        OppiaMetricLog.CurrentScreen.HOME_SCREEN,
+        createStartupLatencyLoggableMetric(startupLatency)
+      )
+    }
+  }
+
+  /** Logs the memrory usage of the application as a performance metric. */
+  fun logMemoryUsage() {
+    performanceMetricsController.logMediumPriorityMetricEvent(
+      oppiaClock.getCurrentTimeMs(),
+      OppiaMetricLog.CurrentScreen.SCREEN_UNSPECIFIED,
+      createMemoryUsageLoggableMetric(performanceMetricsUtils.getTotalPssUsed())
+    )
+  }
+
+  /** Logs the network usage of the application as a performance metric. */
+  fun logNetworkUsage() {
+    performanceMetricsController.logHighPriorityMetricEvent(
+      oppiaClock.getCurrentTimeMs(),
+      OppiaMetricLog.CurrentScreen.SCREEN_UNSPECIFIED,
+      createNetworkUsageLoggableMetric(
+        performanceMetricsUtils.getTotalReceivedBytes(),
+        performanceMetricsUtils.getTotalSentBytes()
+      )
+    )
+  }
+
+  /** Logs the cpu usage of the application as a performance metric. */
+  fun logCpuUsage(cpuUsage: Long) {
+    performanceMetricsController.logHighPriorityMetricEvent(
+      oppiaClock.getCurrentTimeMs(),
+      OppiaMetricLog.CurrentScreen.SCREEN_UNSPECIFIED,
+      createCpuUsageLoggableMetric(cpuUsage)
+    )
+  }
+
+  internal companion object {
+    /**
+     * Logs a high priority performance metric occurring at [currentScreen] defined by [loggableMetric].
+     *
+     * This will schedule a background upload of the event if there's internet connectivity, otherwise
+     * it will cache the event for a later upload.
+     *
+     * This method should only be used for logging periodic metrics like network and cpu usage of the
+     * application. These metrics are important to log and should be prioritized over metrics logged
+     * via [logMediumPriorityMetricEvent] and [logLowPriorityMetricEvent].
+     */
+    internal fun PerformanceMetricsController.logHighPriorityMetricEvent(
```

Comment:
Why are these internal rather than private? I'm not seeing why these should be part of the public API of this class.

Reply:
done.